pkg/instrumentation/auto/util.go

```diff
@@ -0,0 +1,101 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package auto
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/go-logr/logr"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/aws/amazon-cloudwatch-agent-operator/pkg/instrumentation"
+)
+
+// configureAutoAnnotation handles the auto annotation configuration logic
+func configureAutoAnnotation(autoAnnotationConfigStr string, client client.Client, reader client.Reader, setupLog logr.Logger) (InstrumentationAnnotator, error) {
+	// Check environment variables first
+	if os.Getenv("DISABLE_AUTO_ANNOTATION") == "true" {
+		setupLog.Info("detected DISABLE_AUTO_ANNOTATION environment variable, disabling AutoAnnotation")
+		return nil, nil
+	}
+
+	if autoAnnotationConfigStr == "" {
+		return nil, fmt.Errorf("auto-annotation configuration not provided, disabling AutoAnnotation")
```

Comment:
Why does this get logged as an error when the environment variable case is logged as an info message? This seems like a common case.

Reply:
Agree that this probably doesnt need to be an error.
But dont think this will be very common since our add-on/chart will set the default for this as the `java: {deployments:[], ...}` string.

Reply:
Actually we do the same below for `autoAnnotationConfig.Empty()` and that will be true.